codegen/src/main/twirl/templates/ScalaClient/Client.scala.txt

```diff
@@ -18,8 +18,19 @@ import scala.concurrent.{ ExecutionContext, Future, Promise }
 
 import io.grpc._
 
-final class @{service.name}Client(settings: GrpcClientSettings)(implicit mat: Materializer, ex: ExecutionContext) extends @{service.name} with AkkaGrpcClient {
-  import @{service.name}Client._
+// Not sealed so users can extend to write their stubs
+trait @{service.name}Client extends @{service.name} with @{service.name}ClientPowerApi with AkkaGrpcClient
+
+object @{service.name}Client {
+  def apply(settings: GrpcClientSettings)(implicit mat: Materializer, ex: ExecutionContext): @{service.name}Client =
+    new Default@{service.name}Client(settings)
+  // A factory method use by some tooling to reflectively create a @{service.name}Client.
+  def create(settings: GrpcClientSettings, mat: Materializer, ex: ExecutionContext): @{service.name}Client =
```

Comment:
Hmm so it's not possible to discover the apply method in the same way? interesting

Reply:
TBH, I haven't tried. I first attempted to make it work consolidating the `XService.create(settings, mat, ex)` idiom for both scala and java.

But it works indeed. I'll simplify...